src/main/java/io/xstefank/wildfly/bot/ConfigFileChangeProcessor.java

```diff
@@ -0,0 +1,103 @@
+
+    void onFileChanged(@PullRequest.Opened @PullRequest.Edited GHEventPayload.PullRequest pullRequestPayload) {
+        GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
+        for (GHPullRequestFileDetail changedFile : pullRequest.listFiles()) {
+            if (changedFile.getFilename().equals(fileProvider.getFilePath(RuntimeConstants.CONFIG_FILE_NAME))) {
+                String stringPatch = adjustDiffFormat(changedFile.getPatch());
+                List<String> lines = Arrays.asList(stringPatch.split("\n"));
+                Patch<String> patch = UnifiedDiffUtils.parseUnifiedDiff(lines);
+
+                String currentFile = fileProvider.fetchConfigFile(pullRequestPayload.getRepository(), RuntimeConstants.CONFIG_FILE_NAME, ConfigFile.Source.DEFAULT, String.class).get();
+                List<String> currentFileLines = Arrays.asList(currentFile.split("\n"));
+                try {
+                    String updatedFile = patch.applyTo(currentFileLines)
+                            .stream()
+                            .collect(Collectors.joining("\n"));
+                    Optional<WildFlyConfigFile> file = Optional.ofNullable(yamlObjectMapper.readValue(updatedFile, WildFlyConfigFile.class));
+                    List<String> invalidRules = validateFile(file.get());
+
+                    if (invalidRules.isEmpty()) {
+                        updateFormatCommitStatus(pullRequest, GHCommitState.SUCCESS, "\u2705");
+                        Log.debug("Configuration File check successful");
+                    } else {
+                        updateFormatCommitStatus(pullRequest, GHCommitState.ERROR, "\u274C " + String.join(", ", invalidRules));
+                        Log.debugf("Configuration File check unsuccessful [%s]", String.join(",", invalidRules));
+                    }
+                } catch (IOException | PatchFailedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+    }
+
+    List<String> validateFile(WildFlyConfigFile wildflyConfigFile) {
+        List<String> invalidRules = new ArrayList<>();
+        for (WildFlyConfigFile.WildFlyRule rule : wildflyConfigFile.wildfly.rules) {
+            if (rule.id == null) {
+                invalidRules.add("Invalid rule: " + rule);
+            }
+        }
+        return invalidRules;
+    }
+
+    private String adjustDiffFormat(String patch) {
+        Matcher matcher = DIFF_REGEX.matcher(patch);
+        if (!matcher.find()) {
+            throw new IllegalStateException("Unable to find parse diff format");
+        }
+        String result = "+++\n" + (matcher.group(1).startsWith("\n")
+                ? patch
+                : matcher.group(0) + "\n" + patch.substring(matcher.group(0).length() + 1));
+
+        return result;
+    }
+
+    private void updateFormatCommitStatus(GHPullRequest pullRequest, GHCommitState commitState, String description) throws IOException {
```

Comment:
we should probably create some utility that we can reuse in all processors

Reply:
I will do it in this PR then.